mersenne-31/src/x86_64_avx512/packing.rs

```diff
@@ -396,6 +469,30 @@ impl AbstractField for PackedMersenne31AVX512 {
     fn generator() -> Self {
         Mersenne31::generator().into()
     }
+
+    #[must_use]
+    #[inline(always)]
+    fn exp_const_u64<const POWER: u64>(&self) -> Self {
+        match POWER {
+            0 => Self::one(),
+            1 => *self,
+            2 => self.square(),
+            3 => self.cube(),
+            4 => self.square().square(),
+            5 => unsafe {
+                let val = self.to_vector();
+                Self::from_vector(exp5(val))
+            },
+            6 => self.square().cube(),
+            7 => {
+                let x2 = self.square();
+                let x3 = x2 * *self;
+                let x4 = x2.square();
+                x3 * x4
+            }
+            _ => self.exp_u64(POWER),
```

Comment:
You could make some of these faster with the incomplete reduction tricks, but I'll let you decide whether this is in scope/worth it.

Reply:
I think we so rarely use these outside of `x -> x^5` that it's not worth it.

I'll add a comment above the function saying that.